### ISSUE TITLE
Crash on auto fullscreen android

### ISSUE THREAD

**Issue by choozn** [jepiqueau/capacitor-video-player#140]
The player crashes sometimes with various media files while starting.

`java.lang.NullPointerException: Attempt to invoke virtual method 'void com.google.android.exoplayer2.ext.cast.CastPlayer.release()' on a null object reference
                                                                                                    	at com.jeep.plugin.capacitor.capacitorvideoplayer.FullscreenExoPlayerFragment.releasePlayer(FullscreenExoPlayerFragment.java:730)
                                                                                                    	at com.jeep.plugin.capacitor.capacitorvideoplayer.FullscreenExoPlayerFragment.onDestroy(FullscreenExoPlayerFragment.java:679)
                                                                                                    	at androidx.fragment.app.Fragment.performDestroy(Fragment.java:3356)
                                                                                                    	at androidx.fragment.app.FragmentStateManager.destroy(FragmentStateManager.java:781)
                                                                                                    	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:335)
                                                                                                    	at androidx.fragment.app.SpecialEffectsController$FragmentStateManagerOperation.complete(SpecialEffectsController.java:771)
                                                                                                    	at androidx.fragment.app.SpecialEffectsController$Operation.cancel(SpecialEffectsController.java:615)
                                                                                                    	at androidx.fragment.app.SpecialEffectsController.forceCompleteAllOperations(SpecialEffectsController.java:350)
                                                                                                    	at androidx.fragment.app.SpecialEffectsController.executePendingOperations(SpecialEffectsController.java:272)
                                                                                                    	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1914)
                                                                                                    	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1819)
                                                                                                    	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1762)
                                                                                                    	at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:547)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:938)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loop(Looper.java:254)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8243)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:612)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1006)`

**Comment by jepiqueau** [comment]
@choozn what was the type a video when it failed? Where you casino the video

**Comment by choozn** [comment]
I'll try to figure out some more details on why exactly the crash happens and then post them below.

**Comment by PhantomPainX** [comment]
> +1
> 
> `FATAL EXCEPTION: main Process: com.ProComMedia.MirabelleTV, PID: 4467 java.lang.IllegalStateException: The fully qualified name of the implementation of OptionsProvider must be provided as a metadata in the AndroidManifest.xml with key com.google.android.gms.cast.framework.OPTIONS_PROVIDER_CLASS_NAME. at com.google.android.gms.cast.framework.CastContext.zzi(com.google.android.gms:play-services-cast-framework@@21.3.0:6) at com.google.android.gms.cast.framework.CastContext.getSharedInstance(com.google.android.gms:play-services-cast-framework@@21.3.0:15) at com.jeep.plugin.capacitor.capacitorvideoplayer.FullscreenExoPlayerFragment.initializeCastService(FullscreenExoPlayerFragment.java:1262) at 
[...]
>

@stephanevallejo 

Hey, this is due to a lack of an implementation in your manifest, in fact, the error tells you this: "The fully qualified name of the implementation of OptionsProvider must be provided as a metadata in the AndroidManifest.xml with key com.google.android.gms.cast.framework.OPTIONS_PROVIDER_CLASS_NAME"

Read the documentation: https://github.com/jepiqueau/capacitor-video-player/blob/master/docs/API.md#android-quirks

**Comment by stephanevallejo** [comment]
Thanks, it works but when I turn the phone (in landscape mode) the video doesn't go to full screen.

**Comment by PhantomPainX** [comment]
> Thanks, it works but when I turn the phone (in landscape mode) the video doesn't go to full screen.

The display mode can be set with "displayMode" option in initPlayer. If you are using the last version of the plugin, the default one is "all", it means that the screen follows the orientation of the device.

You can see it in the definitions: https://github.com/jepiqueau/capacitor-video-player/blob/master/dist/esm/definitions.d.ts
/**
 * Display Mode ["all", "portrait", "landscape"] (iOS, Android)
 * default: "all"
*/

```
openVideo() {
    CapacitorVideoPlayer.initPlayer({
      mode: "fullscreen",
      pipEnabled: true,
      displayMode: "landscape",
      bkmodeEnabled: true,
      playerId: "live",
      url: "https://XXXXXX/playlist.m3u8"
    });
}
```
  }

**Comment by stephanevallejo** [comment]
I use capacitor-video-player@5.0.2

Without the "displayMode" parameter, on iOS the video turns when I turn the phone.
Without the "displayMode" parameter, on Android the video does not rotate when I turn the phone.

I added the "displayMode" parameter to "all" on Android but the video doesn't turn when I turn the phone.
If I set the "displayMode" parameter to "landscape" on Android, the video automatically rotates in landscape, but there's no portrait-landscape toggling possible.
